remoting/subsystem/src/main/java/org/jboss/as/remoting/HttpListenerRegistryService.java

```diff
@@ -2,44 +2,66 @@
 
 import io.undertow.server.ListenerRegistry;
 import org.jboss.msc.service.Service;
+import org.jboss.msc.service.ServiceBuilder;
 import org.jboss.msc.service.ServiceName;
 import org.jboss.msc.service.ServiceTarget;
 import org.jboss.msc.service.StartContext;
-import org.jboss.msc.service.StartException;
 import org.jboss.msc.service.StopContext;
 
+import java.util.function.Consumer;
+
 /**
  * Service that maintains a registry of all Undertow listeners, and the services that are registered on them.
  *
  * TODO: not sure if this really belongs here conceptually, but in practice it is only used to match upgrade handlers with listeners
  *
  * @author Stuart Douglas
+ * @author <a href="mailto:ropalka@redhat.com">Richard Opalka</a>
  */
 public class HttpListenerRegistryService implements Service<ListenerRegistry> {
 
     @Deprecated
     public static final ServiceName SERVICE_NAME = ServiceName.JBOSS.append("http", "listener", "registry");
 
+    private final Consumer<ListenerRegistry> serviceConsumer;
```

Comment:
Could using an `Optional<Consumer<ListenerRegistry>>` be an alternative?

Reply:
Yes, Optional is an alternative. But it depends on developer's preference. I'm not big fan of Optional.